zap/src/main/java/org/zaproxy/zap/view/HttpPanelManager.java

```diff
@@ -56,68 +61,95 @@ private static synchronized void createInstance() {
     }
 
     public void addRequestPanel(HttpPanel panel) {
-        requestPanels.addPanel(panel);
+        runInEdt(() -> requestPanels.addPanel(panel));
+    }
+
+    private static void runInEdt(Runnable runnable) {
```

Comment:
Is it worth moving this to a public utils type class so that it can be used everywhere?
We need to do it in so many places...

Reply:
There's `ThreadUtils` but does not handle the exceptions (I guess we could still use it here).

Reply:
Moved the method that handles the exceptions to `ThreadUtils` and changed this class to use it.